src/utils/treeUtils.ts

```diff
@@ -930,25 +930,30 @@ export function useTreeSearch<T>(props: TreeSearchProps<T>) {
   );
 
   // Use search keywords to filter options.
-  const [searchKeywordState, setSearchKeyword] = useState(() => searchKeyword ?? '');
+  const [searchKeywordState, setSearchKeyword] = useState(searchKeyword ?? '');
   const [filteredData, setFilteredData] = useState(() =>
     filterVisibleData(data, searchKeywordState)
   );
 
+  const handleSearch = (searchKeyword: string, event?: React.ChangeEvent) => {
+    const filteredData = filterVisibleData(data, searchKeyword);
+    setFilteredData(filteredData);
+    setSearchKeyword(searchKeyword);
+    event && callback?.(searchKeyword, filteredData, event);
+  };
+
+  useEffect(() => {
+    handleSearch(searchKeyword ?? '');
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchKeyword]);
+
```

Comment:
Why not derive `filteredData` from `data` and `searchKeywordState`?

Reply:
You are right. `filtereredData` should be derived from `data` and `searchKeywordState`.

As the useTreeSearch hook is utilized by both the CheckTree and Tree components, I intend to create a new Pull Request with the goal of optimizing it.